apps/api/src/services/data-provider/data-provider.service.ts

```diff
@@ -667,7 +667,9 @@ export class DataProviderService {
 
     for (const date in rootData) {
       data[date] = {
-        marketPrice: new Big(factor).mul(rootData[date].marketPrice).toNumber()
+        marketPrice: rootData[date].marketPrice
```

Comment:
Should be check for `rootData[date].marketPrice >= 0`? Currently, `0` would result in `null`.

Reply:
I assumed if the marketPrice is 0 there might be a problem anyway.
But you are right, I even went a step further allowing negative marketPrices an strictly checking for null values.

Reply:
Looks like we lost the factor now. Can you check my change please?